cls/SourceControl/Git/Utils.cls

```diff
@@ -327,7 +327,10 @@ ClassMethod AfterUserAction(Type As %Integer, Name As %String, InternalName As %
         }
     } elseif (menuItemName = "SwitchBranch") {
         if (Answer = 1) {
-            do ..SwitchBranch(Msg)
+            set status = ..SwitchBranch(Msg)
+            if $$$ISERR(status) {
+                Quit status
```

Comment:
Quitting with an error %Status is the right approach if something goes wrong in AfterUserAction; short of changing to a CSP page like sync I don't see a way to get a better UX than with this.